In [1]:
import json
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
file = '/Users/forheart/Downloads/yelp_dataset/yelp_academic_dataset_business.json'

In [3]:
max_records = 1e5
df = pd.read_json(file, lines=True, chunksize=max_records)
data = pd.DataFrame()
try:
   for df_chunk in df:
       data = pd.concat([data, df_chunk])
except ValueError:
       print ('\nSome messages in the file cannot be parsed')

In [4]:
data.shape

(188593, 15)

In [5]:
data.head()

address                                         attributes  \
0    1314 44 Avenue NE  {'BikeParking': 'False', 'BusinessAcceptsCredi...   
1                       {'Alcohol': 'none', 'BikeParking': 'False', 'B...   
2  1335 rue Beaubien E  {'Alcohol': 'beer_and_wine', 'Ambience': '{'ro...   
3      211 W Monroe St                                               None   
4  2005 Alyth Place SE             {'BusinessAcceptsCreditCards': 'True'}   

              business_id                                         categories  \
0  Apn5Q_b6Nz61Tq4XzPdf9A  Tours, Breweries, Pizza, Restaurants, Food, Ho...   
1  AjEbIBw6ZFfln7ePHha9PA  Chicken Wings, Burgers, Caterers, Street Vendo...   
2  O8S5hYJ1SMc8fA4QBtVujA  Breakfast & Brunch, Restaurants, French, Sandw...   
3  bFzdJJ3wp3PZssNEsyU23g                      Insurance, Financial Services   
4  8USyCYqpScwiNEb58Bt6CA  Home & Garden, Nurseries & Gardening, Shopping...   

        city                                              hours  is_open  \
0    Calgary  {'Monday': '8:30-17:0', 'Tuesday': '11:0-21:0'...        1   
1  Henderson  {'Friday': '17:0-23:0', 'Saturday': '17:0-23:0...        0   
2   Montréal  {'Monday': '10:0-22:0', 'Tuesday': '10:0-22:0'...        0   
3    Phoenix                                               None        1   
4    Calgary  {'Monday': '8:0-17:0', 'Tuesday': '8:0-17:0', ...        1   

    latitude   longitude                  name               neighborhood  \
0  51.091813 -114.031675  Minhas Micro Brewery                              
1  35.960734 -114.939821   CK'S BBQ & Catering                              
2  45.540503  -73.599300         La Bastringue  Rosemont-La Petite-Patrie   
3  33.449999 -112.076979       Geico Insurance                              
4  51.035591 -114.027366         Action Engine                              

  postal_code  review_count  stars state  
0     T2E 6L6            24    4.0    AB  
1       89002             3    4.5    NV  
2     H2G 1K7             5    4.0    QC  
3       85003             8    1.5    AZ  
4     T2H 0N5             4    2.0    AB

The goal is to select the businesses that are restaurants.

In [6]:
restaurants = data[data.categories.str.contains('Restaurants', na=False)][['business_id']]

In [7]:
restaurants.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 57173 entries, 0 to 188586
Data columns (total 1 columns):
business_id    57173 non-null object
dtypes: object(1)
memory usage: 893.3+ KB


In [8]:
# randomly sample 2000 restaurants.
restaurants = restaurants.sample(1000, random_state=1)

In [9]:
# Save this restaurants dataframe as a csv file
restaurants.to_csv('yelp_restaurants')

In [10]:
file1 = '/Users/forheart/Downloads/yelp_dataset/yelp_academic_dataset_review.json'

In [11]:
max_records = 1e5
df1 = pd.read_json(file1, lines=True, chunksize=max_records)
data1 = pd.DataFrame()
try:
   for df_chunk in df1:
       data1 = pd.concat([data1, df_chunk])
except ValueError:
       print ('\nSome messages in the file cannot be parsed')

In [13]:
df_business = pd.DataFrame({'business_id':bybusiness.index, 'review_counts':bybusiness.values})

In [18]:
# Create a dataframe that contains only restaurants and these restaurants have reviews in the review dataframe.
df_res = pd.merge(restaurants, data1, on='business_id', how='left')

In [19]:
df_res.shape

(60788, 9)

In [20]:
df_res.head()

business_id  cool       date  funny               review_id  \
0  xSs1Z1OrWC_KmhY0SPKJ1Q     0 2016-06-01      0  DsPX83pa6vVSASrHF0U1aQ   
1  xSs1Z1OrWC_KmhY0SPKJ1Q     0 2015-05-06      0  HGw5_lZQ3lpnRHFmbK0-Ww   
2  xSs1Z1OrWC_KmhY0SPKJ1Q     0 2016-10-30      0  VsH5hnYRbksrYnH4Vi-ZSg   
3  xSs1Z1OrWC_KmhY0SPKJ1Q     0 2010-06-17      0  xSRTe2zjKjzYqyC-0EThSQ   
4  xSs1Z1OrWC_KmhY0SPKJ1Q     0 2015-10-22      0  3gctU9pG-ikd8XVW58p_wg   

   stars                                               text  useful  \
0      2  My server was polite but not very prompt. The ...       0   
1      4  The food was good and the service was great! W...       1   
2      3  I like to get the habachi vegetables or chicke...       0   
3      2  My daughter and I decided to try Shiki one nig...       1   
4      5  My husband and I don't eat out frequently beca...       0   

                  user_id  
0  ubgsST240v6cc5C7Kt4zKQ  
1  xdcMl8ghySzhDgKzc7OPtQ  
2  rzY8uwz-JtfLowr2EcHo1A  
3  A2l6pDAwA5SkfuOd7E_npw  
4  GljH-HX4zGc-c87NGBDhSQ

There are 188593 businesses in the yelp_academic_dataset_review dataset. There are 57173 businesses that are restaurants. We need to extract the restaurants from the review dataset by filtering the business ids in the restaurants dataframe.

In [22]:
# Create the dataframe of restaurant reviews
df_res.to_csv('res_review')